types/tables.d.ts

```diff
@@ -53,10 +53,17 @@ export interface ArcTable<Item = unknown> {
   update(params: UpdateParams<Item>, callback: Callback<UpdateOutput>): void;
 }
 
-export type ArcDB<Tables> = {
+type ArcDBWith<Tables> = {
   [tableName in keyof Tables]: ArcTable<Tables[tableName]>;
 };
 
+export type ArcDB<Tables> = ArcDBWith<Tables> & {
+  name(name: string): string;
```

Comment:
The `name` is supposed to be the name of a table, right? So instead of `string`, could it be `name:  keyof Tables`? Then in the `itEnforcesTableNames` test, you could add the following assertions:

```
  // $ExpectError
  db.name("wrong");

  // $ExpectType string
  db.name("note");
```

Reply:
Tested this locally and it works. It also makes my IDE auto-suggest the string values which is nice..